flopy/discretization/modeltime.py

```diff
@@ -71,21 +226,373 @@ def totim(self):
+                "and '-'"
+            )
+
+        # check for time component
+        if "t" in str_datetime:
+            dtsep = "t"
+        elif " " in str_datetime:
+            dtsep = " "
+        else:
+            dtsep = None
+
+        # check if year first (yr, month, day) combo...
+        year_first = False
+        tmp = str_datetime.split(dsep)[0]
+        if len(tmp) == 4:
+            year_first = True
+
+        if dtsep is not None:
+            if year_first:
+                str_rep = f"%Y{dsep}%m{dsep}%d{dtsep}%H:%M:%S"
+            else:
+                str_rep = f"%m{dsep}%d{dsep}%Y{dtsep}%H:%M:%S"
+
+        else:
+            if year_first:
+                str_rep = f"%Y{dsep}%m{dsep}%d"
+            else:
+                str_rep = f"%m{dsep}%d{dsep}%Y"
+
+        return str_rep
+
+    @staticmethod
+    def datetime_from_user_input(datetime_obj):
+        """
+        Method to create a datetime.datetime object from a variety of user
+        inputs including the following:
+
+        datetime.datetime objects
+        numpy.datetime64 objects
+        pandas.Timestamp objects
+        string objects
+
+        Supported formats for string objects representing November 12th, 2024
+        are as follows:
+
+        '11/12/2024'
+        '11-12-2024'
+        '2024/11/12'
+        '2024-11-12'
+
+        Time can also be represented in the string object. Example formats
+        representing 2:31 pm on November 12th, 2024 are as follows:
+
+        '2024-11-12T14:31:00'
+        '2024/11/12T14:31:00'
+        '11-12-2024t14:31:00'
+        '11/12/2024t14:31:00'
+        '2024-11-12 14:31:00'
+        '2024/11/12 14:31:00'
+        '11-12-2024 14:31:00'
+        '11/12/2024 14:31:00'
+
+        Parameters
+        ----------
+        datetime_obj : various formats
+            a user supplied representation of date or datetime
+
+        Returns
+        -------
+            datetime.datetime object
+        """
+        if datetime_obj is None:
+            datetime_obj = datetime.datetime(1970, 1, 1)  # unix time zero
+        elif isinstance(datetime_obj, np.datetime64):
+            unix_time_0 = datetime.datetime(1970, 1, 1)
+            ts = (datetime_obj - np.datetime64(unix_time_0)) / np.timedelta64(1, "s")
+            datetime_obj = datetime.datetime.utcfromtimestamp(ts)
+        elif isinstance(datetime_obj, pd.Timestamp):
+            datetime_obj = datetime_obj.to_pydatetime()
+        elif isinstance(datetime_obj, datetime.datetime):
+            pass
+        elif isinstance(datetime_obj, str):
+            str_rep = ModelTime._get_datetime_string_format(datetime_obj)
+            datetime_obj = datetime.datetime.strptime(datetime_obj, str_rep)
+
+        else:
+            raise NotImplementedError(
+                f"{type(datetime_obj)} date representations "
+                f"are not currently supported"
+            )
+
+        return datetime_obj
+
+    def get_totim(self, kper, kstp=None):
```

Comment:
How about `get_model_time` or `get_end_time`?

Reply:
how about `get_kstp_kper_totim()`, since I think that's what it's doing? Also I think it's important to note that simulation time might be different from elapsed time, for example if the simulation starts with a steady-state stress period of some length.